libs/vkd3d/command.c

```diff
@@ -18551,17 +18529,42 @@ static void STDMETHODCALLTYPE d3d12_command_queue_ExecuteCommandLists(ID3D12Comm
     sub.execute.num_command_allocators = command_list_count;
     sub.execute.retained_resources = retained_resources;
     sub.execute.num_retained_resources = num_retained_resources;
+    for (i = 0; i < num_retained_resources; i++)
+    {
+        if (!vkd3d_atomic_uint32_load_explicit(&retained_resources[i]->internal_refcount, vkd3d_memory_order_relaxed) ||
+                d3d12_resource_incref(retained_resources[i]) == 1)
+        {
+            ERR("Trying to execute resource which has already been destroyed. This will likely hang GPU at some point ... Skipping ECL.\n");
+            /* TODO: We could mark device as lost, or just ignore this submit. Hard to tell
```

Comment:
Does Mafia hit this path? Sounds like dropping the device would break the game?

Reply:
It does not. I added this mostly for completeness. Maybe it's sufficient to just detect if the load is 0 and error out, and not try to do any weird shenanigans with rewriting the refcount?